crates/uv/src/commands/python/install.rs

```diff
@@ -354,32 +354,53 @@ pub(crate) async fn install(
-                            .copied()
-                            .chain(existing_installations.iter()),
-                        &target,
-                    );
+                    // Check if the existing link is valid
+                    let valid_link = match target.read_link().and_then(|target| target.try_exists())
+                    {
+                        Ok(valid) => valid,
+                        Err(err) => {
+                            debug!("Failed to inspect executable with error: {err}");
+                            true // Treat this as a valid link, we won't remove it without `--force`
+                        }
+                    };
+
+                    // Figure out what installation it references, if anyf
+                    let existing = valid_link
+                        .then(|| {
+                            find_matching_bin_link(
+                                installations
+                                    .iter()
+                                    .copied()
+                                    .chain(existing_installations.iter()),
+                                &target,
+                            )
+                        })
+                        .flatten();
 
                     match existing {
                         None => {
                             // There's an existing executable we don't manage, require `--force`
-                            if !force {
-                                errors.push((
-                                    installation.key(),
-                                    anyhow::anyhow!(
-                                        "Executable already exists at `{}` but is not managed by uv; use `--force` to replace it",
-                                        to.simplified_display()
-                                    ),
-                                ));
-                                continue;
+                            if valid_link {
+                                if !force {
+                                    errors.push((
+                                        installation.key(),
+                                        anyhow::anyhow!(
+                                            "Executable already exists at `{}` but is not managed by uv; use `--force` to replace it",
+                                            to.simplified_display()
+                                        ),
+                                    ));
+                                    continue;
+                                }
+                                debug!(
+                                    "Replacing existing executable at `{}` due to `--force`",
+                                    target.simplified_display()
+                                );
+                            } else {
+                                debug!(
+                                    "Replacing existing executable at `{}` due to an invalid symlink",
```

Comment:
Maybe "Replacing invalid symlink at `{}`"?

Reply:
Hm I was sticking with the existing style but I do like that wording.